### PR TITLE
Contact and Team Models Remaining Work

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: #230 
# What's Changed

Continuing on with @glennmatthews 's work for #230

# TODO

- [ ] Need a way to Bulk Add Contacts and Teams to Multiple Objects (Might be out of scope?)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)

### CODE REVIEW

File: nautobot/core/forms/forms.py [hunk 1]
```diff
@@ -56,6 +59,120 @@ def clean(self):
         self.instance.address = self.cleaned_data.get("address")
 
 
+class BaseInlineGFKFormSet(forms.BaseModelFormSet):
```
Comment (glennmatthews) [inline nautobot/core/forms/forms.py:62]: I was talking with John this afternoon and he brought up the idea of an alternate approach - instead of this "SecretsGroup-like" approach where the contact/team association to an object is done in the object's edit view via an embedded formset, maybe we should consider a "device-interfaces-like" approach where adding a new ContactAssociation is done from the "associated contacts" table on an object detail view, i.e. with "Add a Contact" or "Add a Team Contact" button/dropdown in the detail view or a dedicated tab therein.

File: nautobot/core/templates/generic/object_create.html [hunk 2]
```diff
@@ -9,8 +10,8 @@
             {{ field }}
         {% endfor %}
         <div class="row">
-            <div class="col-md-6 col-md-offset-3">
-                <h3>
+            <div class="col-md-10 col-md-offset-1 col-lg-8 col-lg-offset-2">
```
Comment (glennmatthews) [inline nautobot/core/templates/generic/object_create.html:13]: Probably should revert the changes here and below and plan to address them more holistically as part of #5045 and/or #5062.

File: nautobot/core/templates/generic/object_retrieve.html [hunk 3]
```diff
@@ -132,6 +132,9 @@ <h1>
             <div class="row">
                 <div class="col-md-12">
                     {% block content_full_width_page %}{% endblock content_full_width_page %}
+                    {% if associated_contacts_table %}
+                        {% include 'panel_table.html' with table=associated_contacts_table %}
+                    {% endif %}
```
Comment (glennmatthews) [inline nautobot/core/templates/generic/object_retrieve.html:137]: Probably we should move this to a separate tab instead of the main tab?

File: nautobot/core/forms/forms.py (hunk 1 above)
Comment (HanlinMiao) [reply nautobot/core/forms/forms.py:62]: This is the pattern we are talking about here right? 
Screen Shot 2024-01-08 at 4 14 33 PM

Comment (HanlinMiao) [reply nautobot/core/forms/forms.py:62]: One thing I can think of that can differentiate the Contact/team feature from this approach is adding an existing contact/team versus creating and then adding a **new** contact/team.

I believe for device it is always adding a **new** interface/console port and etc.

Comment (glennmatthews) [reply nautobot/core/forms/forms.py:62]: Good thought, could be either case ideally I think.

Yeah, what you screenshotted there, plus also:

image

where we have the ability to do CRUD operations directly on the related-objects table.

File: nautobot/extras/templates/extras/secretsgroup_edit.html [hunk 4]
```diff
@@ -55,7 +55,7 @@
                                             {% for error in field.errors %}
                                                 {# Embed an HTML comment indicating the error for extraction by tests #}
                                                 <!-- FORM-ERROR {{ field.name }}: {{ error }} -->
-                                                <li clas="text-danger">{{ error }}</li>
+                                                <li class="text-danger">{{ error }}</li>
```
Comment (itdependsnetworks) [inline nautobot/extras/templates/extras/secretsgroup_edit.html:58]: Same found: https://github.com/nautobot/nautobot/blob/1907613e6b1eff726b6ed78a45f0a18b58cea752/nautobot/extras/templates/extras/dynamicgroup_edit.html#L123

File: nautobot/core/views/mixins.py [hunk 5]
```diff
@@ -232,6 +233,7 @@ class NautobotViewSetMixin(GenericViewSet, AccessMixin, GetReturnURLMixin, FormV
     # serializer_class has to be specified to eliminate the need to override retrieve() in the RetrieveModelMixin for now.
     serializer_class = None
     table_class = None
+    notes_form_class = NoteForm
```
Comment (glennmatthews) [inline nautobot/core/views/mixins.py]: Why is this needed? I did a `git grep notes_form_class` and didn't see anywhere that it was being used - what'd I miss?

File: nautobot/core/templates/generic/object_retrieve.html [hunk 6]
```diff
@@ -86,6 +108,13 @@ <h1>
                 Advanced
             </a>
         </li>
+        {% if associated_contacts_table %}
+            <li role="presentation"{% if request.GET.tab == 'contacts' %} class="active"{% endif %}>
+                <a href="{{ object.get_absolute_url }}#contacts" onclick="switch_tab(this.href)" aria-controls="contacts" role="tab" data-toggle="tab">
+                    Contacts/Teams
```
Comment (glennmatthews) [inline nautobot/core/templates/generic/object_retrieve.html:108]: Maybe not for this PR, but John and I were bouncing around the idea of using mdi icons for some of the detail view tabs instead of text so as to mitigate tab sprawl on views like Device detail, i.e. instead of: 

image

something like:

image

File: nautobot/core/templates/generic/object_retrieve.html [hunk 7]
```diff
@@ -154,6 +183,45 @@ <h1>
...
+                                    </div>
+                                </div>
+                                {% render_table associated_contacts_table 'inc/table.html' %}
+                                <div class="panel-footer noprint">
+                                    {% if perms.dcim.delete_contactassociation %}
+                                        <button type="submit" formaction="{% url 'extras:contactassociation_bulk_delete' %}?return_url={{request.path}}" class="btn btn-danger btn-xs">
+                                            <span class="mdi mdi-trash-can-outline" aria-hidden="true"></span> Delete
+                                        </button>
+                                    {% endif %}
+                                    {% if perms.dcim.add_contactassociation %}
```
Comment (glennmatthews) [inline nautobot/core/templates/generic/object_retrieve.html:202]: Certainly we need to at least be self-consistent about permissions that we're checking :-)

File: nautobot/extras/templates/extras/secretsgroup_edit.html [hunk 8]
```diff
@@ -53,13 +53,23 @@
...
+                                    {{ field }}
+                                    {% if field.errors %}
+                                        <ul>
+                                            {% for error in field.errors %}
+                                                {# Embed an HTML comment indicating the error for extraction by tests #}
+                                                <!-- FORM-ERROR {{ field.name }}: {{ error }} -->
+                                                <li class="text-danger">{{ error }}</li>
+                                            {% endfor %}
+                                        </ul>
+                                    {% endif %} d
```
Comment (itdependsnetworks) [inline nautobot/extras/templates/extras/secretsgroup_edit.html:66]:
```suggestion
                                    {% endif %}
```

File: nautobot/extras/templates/extras/object_assign_contact_or_team.html [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{% extends 'generic/object_edit.html' %}
+{% load static %}
```
Comment (glennmatthews) [inline nautobot/extras/templates/extras/object_assign_contact_or_team.html:2]: Unneeded?

File: nautobot/extras/templates/extras/object_new_contact.html [hunk 10]
```diff
@@ -0,0 +1,34 @@
+{% extends 'generic/object_edit.html' %}
+{% load static %}
```
Comment (glennmatthews) [inline nautobot/extras/templates/extras/object_new_contact.html:2]: Unneeded?

File: nautobot/extras/templates/extras/secretsgroup_edit.html [hunk 11]
```diff
@@ -53,13 +53,23 @@
...
                                             <li clas="text-danger">{{ error }}</li>
                                         {% endfor %}
-                                    </ul>
-                                {% endif %}
-                            </td>
-                        {% endfor %}
-                    </tr>
-                {% endfor %}
-            </table>
+                                    {% endif %}
```
Comment (glennmatthews) [inline nautobot/extras/templates/extras/secretsgroup_edit.html:56]: Are the changes to this file correct/relevant? They look wrong to me. Maybe revert?

File: nautobot/extras/templates/extras/team_retrieve.html [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+            <td>{{ object.name }}</td>
+        </tr>
+        <tr>
+            <td>Phone</td>
+            <td>
+                {% if object.phone %}
+                    <a href="tel:{{ object.phone }}">{{ object.phone }}</a>
+                {% else %}
+                    <span class="text-muted">&mdash;</span>
+                {% endif %}
```
Comment (glennmatthews) [inline nautobot/extras/templates/extras/team_retrieve.html:21]: Maybe a future enhancement, but the `phone` and `email` rendering logic here, below, and elsewhere might be a great candidate for implementing as templatetags?

File: nautobot/extras/tests/test_api.py [hunk 13]
```diff
@@ -376,6 +379,83 @@ def test_get_object_with_or_without_permission(self):
...
+    create_data = []
+
+    @classmethod
+    def setUpTestData(cls):
+        ContactAssociation.objects.create(
+            contact=Contact.objects.first(),
+            associated_object_type=ContentType.objects.get_for_model(IPAddress),
+            associated_object_id=IPAddress.objects.first().pk,
+            status=Status.objects.get_for_model(ContactAssociation).first(),
+        )
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_api.py:428]: Probably at least one of these should have a non-null `role`?

File: nautobot/extras/tests/test_api.py [hunk 14]
```diff
@@ -3464,6 +3544,48 @@ def test_update_tag_content_type_unchanged(self):
...
+#
+
+
+class TeamTest(APIViewTestCases.APIViewTestCase):
+    model = Team
+    create_data = [
+        {
+            "name": "Team 1",
+            "phone": "12345212",
+            "email": "team1@gmail.com",
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_api.py:3558]: Probably best to use "definitively" fake phone numbers (555-01xx) and emails (@example.com) here and in the other tests.

File: nautobot/extras/filters/__init__.py [hunk 15]
```diff
@@ -388,6 +394,23 @@ class NautobotFilterSet(
...
+#
+
+
+class ContactFilterSet(NameSearchFilterSet, NautobotFilterSet):
+    class Meta:
+        model = Contact
+        fields = "__all__"
+
+
+class ContactAssociationFilterSet(NautobotFilterSet):
```
Comment (glennmatthews) [inline nautobot/extras/filters/__init__.py:408]: Probably needs a bespoke `q` filter?

File: nautobot/extras/urls.py [hunk 16]
```diff
@@ -166,6 +169,10 @@
         name="configcontextschema_notes",
         kwargs={"model": ConfigContextSchema},
     ),
+    # contacts
+    path("contact-associations/add-new-contact/", views.ObjectNewContactView.as_view(), name="object_contact_add"),
+    path("contact-associations/add-new-team/", views.ObjectNewTeamView.as_view(), name="object_team_add"),
+    path("contact-associations/assign-contact-team/", views.ObjectAssignView.as_view(), name="object_assign"),
```
Comment (glennmatthews) [inline nautobot/extras/urls.py:175]: Probably would want to rename this to `object_contact_team_assign` or similar as `object_assign` is a bit vague?

File: nautobot/extras/views.py [hunk 17]
```diff
@@ -355,6 +363,216 @@ class ConfigContextSchemaBulkDeleteView(generic.BulkDeleteView):
...
+
+class ObjectNewContactView(generic.ObjectEditView):
+    queryset = Contact.objects.all()
+    model_form = forms.ObjectNewContactForm
+    template_name = "extras/object_new_contact.html"
+
+    def successful_post(self, request, obj, created, new_logger):
+        """Callback after the form is successfully saved but before redirecting the user."""
+        verb = "Created" if created else "Modified"
+        msg = f"{verb} {self.queryset.model._meta.verbose_name} and Created Contact Association"
```
Comment (glennmatthews) [inline nautobot/extras/views.py:416]: This feels to me like maybe it's exposing too much of the system "guts" to the end user. They don't really care that Contact Association is a separate part of the data model? Same comment below.

File: nautobot/extras/views.py [hunk 18]
```diff
@@ -355,6 +363,216 @@ class ConfigContextSchemaBulkDeleteView(generic.BulkDeleteView):
...
+                "obj_type": self.queryset.model._meta.verbose_name,
+                "form": form,
+                "return_url": self.get_return_url(request, obj),
+                "editing": obj.present_in_database,
+                **self.get_extra_context(request, obj),
+            },
+        )
+
+
+class ObjectAssignView(generic.ObjectEditView):
```
Comment (glennmatthews) [inline nautobot/extras/views.py:570]: Here too maybe something like `ObjectAssignContactOrTeamView` might be more explicit.

**Review by glennmatthews — COMMENTED** [review]
Lots of feedback, sorry!

Are we planning to include documentation in this PR or will that be a subsequent one?

File: nautobot/core/management/commands/generate_test_data.py [hunk 19]
```diff
@@ -99,6 +106,10 @@ def _generate_factory_data(self, seed, db_name):
         TagFactory.create_batch(5, content_types=TaggableClassesQuery().as_queryset(), using=db_name)
         # ...and some tags that apply to a random subset of content-types
         TagFactory.create_batch(15, using=db_name)
+        self.stdout.write("Creating Contacts...")
+        ContactFactory.create_batch(20, using=db_name)
+        self.stdout.write("Creating Teams...")
+        TeamFactory.create_batch(5, using=db_name)
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/core/management/commands/generate_test_data.py:112]: Yeah I was trying to make a `ContactAssociationFactory` yesterday, it quickly got complicated with `associated_object_id` and `associated_object_type` and how to populate them. So I decided to move on to things more critical.

File: nautobot/core/templates/generic/object_retrieve.html [hunk 20]
```diff
@@ -151,6 +166,44 @@ <h1>
...
+                                    <strong>Contact Associations</strong>
+                                    <div class="pull-right noprint">
+                                        {% if request.user.is_authenticated %}
+                                            <button type="button" class="btn btn-default btn-xs" data-toggle="modal" data-target="#AssociatedContactsTable_config" title="Configure table"><i class="mdi mdi-cog"></i> Configure</button>
+                                        {% endif %}
+                                    </div>
+                                </div>
+                                {% render_table associated_contacts_table 'inc/table.html' %}
+                                <div class="panel-footer noprint">
+                                    {% if perms.extras.delete_contactassociation %}
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/core/templates/generic/object_retrieve.html:186]: I feel like that kinda opens up a hole where it seems reasonable to use other permissions as well. `can_change:contact` for example.

File: nautobot/core/views/generic.py [hunk 21]
```diff
@@ -67,6 +68,7 @@ class ObjectView(ObjectPermissionRequiredMixin, View):
 
     queryset = None
     template_name = None
+    is_contact_model = True
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/core/views/generic.py:71]: Could be a Team design meeting topic, how are we going to disallow that sort of operation?

File: nautobot/extras/forms/contacts.py [hunk 22]
```diff
@@ -0,0 +1,159 @@
...
+            "tags",
+        ]
+
+
+class ContactBulkEditForm(TagsBulkEditFormMixin, NautobotBulkEditForm):
+    pk = forms.ModelMultipleChoiceField(queryset=Contact.objects.all(), widget=forms.MultipleHiddenInput())
+    address = forms.CharField(required=False, widget=forms.Textarea())
+
+    class Meta:
+        model = Contact
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/extras/forms/contacts.py:38]: I believe it is ready as it is. Am I missing fields that make sense to be able to bulk edit?

File: nautobot/extras/models/tags.py [hunk 23]
```diff
@@ -6,11 +6,13 @@
...
-from nautobot.extras.models import ChangeLoggedModel, CustomFieldModel
-from nautobot.extras.models.mixins import NotesMixin
-from nautobot.extras.models.relationships import RelationshipModel
 from nautobot.extras.utils import extras_features, TaggableClassesQuery
 
+from .change_logging import ChangeLoggedModel
+from .customfields import CustomFieldModel
+from .mixins import NotesMixin
+from .relationships import RelationshipModel
+
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/extras/models/tags.py:15]: I do not think I made this change. I reverted it for now.

[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/extras/models/tags.py:15]: Oops it is a circular import problem lol

File: nautobot/core/views/mixins.py (hunk 5 above)
Comment (HanlinMiao) [reply nautobot/core/views/mixins.py]: It is for this test here
Screen Shot 2024-01-11 at 4 36 35 PM

File: nautobot/extras/api/serializers.py [hunk 24]
```diff
@@ -175,6 +178,23 @@ def get_owner(self, obj):
...
 
 
+#
+# Contacts
+#
+
+
+class ContactSerializer(NautobotModelSerializer):
+    class Meta:
+        model = Contact
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/extras/api/serializers.py:189]: It seems like the serializers, at their current state, allow a Team to set its contacts via the REST API, but not vice versa?

File: nautobot/core/views/mixins.py (hunk 5 above)
Comment (glennmatthews) [reply nautobot/core/views/mixins.py]: Ah, that's why I didn't find it:

```python
            if getattr(self, f"{self.action}_form_class"):
                form_class = getattr(self, f"{self.action}_form_class")
```

File: nautobot/core/templates/generic/object_retrieve.html (hunk 20 above)
[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/core/templates/generic/object_retrieve.html:186]: I agree that we need to reach and document a decision. But right now we're using `can_change:object` in one place in the template and `perms.extras.*_contactassociation` elsewhere as gates on a closely related set of actions, which seems like the worst from a consistency standpoint.

[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/core/templates/generic/object_retrieve.html:186]: Possibly we should just use the contactassociation permissions in each case?

File: nautobot/extras/api/serializers.py (hunk 24 above)
[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/extras/api/serializers.py:189]: that makes sense given the direction of the M2M definition. Would be nice to fix if we can.

File: nautobot/extras/forms/contacts.py (hunk 22 above)
[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/extras/forms/contacts.py:38]: I was thinking that if we support bulk-edit of address, why not also phone and email?

File: nautobot/extras/models/tags.py (hunk 23 above)
[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/extras/models/tags.py:15]: Afraid of that :) Can you add a comment to that effect?

File: nautobot/extras/forms/contacts.py (hunk 22 above)
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/extras/forms/contacts.py:48]: Phone and email fields added

File: nautobot/extras/templates/extras/object_assign_contact_or_team.html [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+        {% include 'extras/inc/object_contact_header.html' with active_tab='assign' %}
+    {% endif %}
+{% endblock %}
+
+{% block form %}
+    <div class="panel panel-default">
+        <div class="panel-heading"><strong>Contact Association</strong></div>
+        <div class="panel-body">
+            {% render_field form.contact %}
+            {% render_field form.team %}
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/extras/templates/extras/object_assign_contact_or_team.html:16]: https://github.com/nautobot/nautobot/assets/46973263/3290b040-0bed-4e9b-9c2d-30c51217cb16
Included in the latest commit.

File: nautobot/core/templates/generic/object_retrieve.html [hunk 26]
```diff
@@ -151,6 +166,44 @@ <h1>
...
+                                        <button type="submit" formaction="{% url 'extras:contactassociation_bulk_delete' %}?return_url={{request.path}}" class="btn btn-danger btn-xs">
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/core/templates/generic/object_retrieve.html:187]: https://github.com/nautobot/nautobot/issues/5105.

File: nautobot/core/templates/generic/object_retrieve.html [hunk 27]
```diff
@@ -151,6 +166,44 @@ <h1>
...
+                    <div class="col-md-12">
+                        <form method="post">
+                            {% csrf_token %}
+                            <div class="panel panel-default">
+                                <div class="panel-heading">
+                                    <strong>Contact Associations</strong>
+                                    <div class="pull-right noprint">
+                                        {% if request.user.is_authenticated %}
+                                            <button type="button" class="btn btn-default btn-xs" data-toggle="modal" data-target="#AssociatedContactsTable_config" title="Configure table"><i class="mdi mdi-cog"></i> Configure</button>
+                                        {% endif %}
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/core/templates/generic/object_retrieve.html:181]: Shoot it does not work. When I click on save on the ConfigForm, it just redirects me to the main tab with the `?columns=`. When I navigate to the Contacts tab, `?columns` just disappears. It seems like all the table config forms that work have their own respective template and view?
Screen Shot 2024-01-16 at 10 09 41 AM

File: nautobot/extras/templates/extras/team_retrieve.html (hunk 12 above)
Comment (HanlinMiao) [reply nautobot/extras/templates/extras/team_retrieve.html:21]: https://github.com/nautobot/nautobot/issues/5107

File: nautobot/core/templates/generic/object_retrieve.html (hunk 27 above)
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/core/templates/generic/object_retrieve.html:181]: Should we hide it for now and document it as future work?

**Review by glennmatthews — APPROVED** [review]

**Comment by glennmatthews** [comment]
approving to keep this moving forward - don't forget documentation though! :-)

File: nautobot/core/templates/generic/object_retrieve.html (hunk 27 above)
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/core/templates/generic/object_retrieve.html]: https://github.com/nautobot/nautobot/issues/5112

File: mkdocs.yml [hunk 28]
```diff
@@ -248,6 +250,7 @@ plugins:
         'plugins/porting-from-netbox.md': 'development/apps/porting-from-netbox.md'
         'rest-api/overview.md': 'user-guide/platform-functionality/rest-api/overview.md'
         'user-guide/administration/configuration/node-configuration.md': 'user-guide/administration/configuration/index.md'
+        'user-guides/contact-and-team.md': 'user-guide/feature-guides/contact-and-team.md'
```
Comment (glennmatthews) [inline mkdocs.yml:253]: Don't really need a redirect here since this isn't a pre-existing documentation page.

File: mkdocs.yml [hunk 29]
```diff
@@ -374,7 +378,9 @@ nav:
         - Virtual Chassis: "user-guide/core-data-model/dcim/virtualchassis.md"
       - Extras:
         - Config Context: "user-guide/core-data-model/extras/configcontext.md"
+        - Contact: "user-guide/core-data-model/extras/contact.md"
         - Config Context Schema: "user-guide/core-data-model/extras/configcontextschema.md"
```
Comment (glennmatthews) [inline mkdocs.yml:382]:
```suggestion
        - Config Context Schema: "user-guide/core-data-model/extras/configcontextschema.md"
        - Contact: "user-guide/core-data-model/extras/contact.md"
```

File: mkdocs.yml [hunk 30]
```diff
@@ -533,6 +539,7 @@ nav:
       - Release Checklist: "development/core/release-checklist.md"
   - Release Notes:
     - "release-notes/index.md"
+    - Version 2.2: "release-notes/version-2.2.md"
```
Comment (glennmatthews) [inline mkdocs.yml:542]: Would you mind also updating the `tool.towncrier` section in `pyproject.toml` to point to this file instead of `version-2.1.md`?

File: nautobot/docs/user-guide/feature-guides/contact-and-team.md [hunk 31]
```diff
@@ -0,0 +1,52 @@
+# Contacts and Teams
+
+## How to Assign Contacts and Teams to Nautobot Objects
+
+In order to assign contacts/teams to a particular object, we first need to navigate to the detail view of the object. We will use a circuit in this example.
```
Comment (glennmatthews) [inline nautobot/docs/user-guide/feature-guides/contact-and-team.md:5]: Would be good perhaps to point out that you need to define relevant Status and Role objects for the `contact association` content-type first?

File: nautobot/docs/user-guide/feature-guides/contact-and-team.md [hunk 32]
```diff
@@ -0,0 +1,52 @@
...
+![Contact Form Tabs](./images/contact-and-team/contact_form_tabs.png)
+
+### Assign a New Contact/Team
+
+Which form to use depends on the use case but in this example, we will use the first form which creates a **new** contact instance and associates it with the object.
+
+1. We first fill out the contact information.
+![Contact Info](./images/contact-and-team/new_contact_info.png)
+2. Then we fill out the information needed for the assignment of this contact to the object.
+![Contact Association Info](./images/contact-and-team/new_contact_association_info.png)
```
Comment (glennmatthews) [inline nautobot/docs/user-guide/feature-guides/contact-and-team.md:30]: Would be nice to use somewhat more realistic data here. Status `Active` is probably fine to use, and I'd experimented with roles like `L1 Support` and `Manager` for contact associations.
